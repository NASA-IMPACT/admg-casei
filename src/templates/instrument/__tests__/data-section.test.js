import React from "react"
import { create } from "react-test-renderer"

import DataSection from "../data-section"
import { instrumentQuery } from "../../../../test/__fixtures__"

describe("Data Section", () => {
  it("matches snapshot", () => {
    const tree = create(
      <DataSection id="data" dois={instrumentQuery.data.instrument.dois} />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
