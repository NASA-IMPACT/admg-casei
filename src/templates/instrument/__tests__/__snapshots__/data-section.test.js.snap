// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Data Section matches snapshot 1`] = `
.c0.c0 {
  display: grid;
  grid-template-columns: repeat(12,1fr);
  -webkit-column-gap: 1rem;
  column-gap: 1rem;
  row-gap: 2rem;
  position: relative;
  margin: 0;
  padding: 0;
  margin-bottom: 6rem;
  background-color: none;
}

.c0.c0 > *,
.c0.c0 h3 {
  color: #e9eaec;
}

.c1.c1 {
  position: absolute;
  top: -80px;
}

.c3.c3 {
  grid-column: 1 / span 12;
}

.c2.c2 {
  grid-column: 1 / span 7;
  -webkit-align-self: end;
  -ms-flex-item-align: end;
  align-self: end;
}

.c4.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  max-height: 35rem;
  overflow: auto;
  gap: 1rem;
}

.c5.c5 {
  display: grid;
  gap: 1rem;
  grid-template-columns: 1fr 1fr;
  background-color: #303641;
  padding: 1.5rem;
}

.c6.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 1rem;
}

<section
  className="c0"
  data-cy="data-section"
  mode="darkTheme"
>
  <div
    className="c1"
    id="data"
  />
  <div
    className="c2"
    data-cy="data-section-header"
  >
    <a
      href="#data"
    >
      <h2>
        Data Products
      </h2>
    </a>
  </div>
  <div
    className="c3"
  >
    <p>
      Filter data products from this instrument by specific campaigns or platforms.
    </p>
    <div
      className="c4"
    >
      <div
        className="c5"
        data-cy="data-product"
      >
        <div>
          <label
            data-cy="doi-label"
          >
            GPM GROUND VALIDATION WYOMING KING AIR CLOUD MICROPHYSICS LPVEX V1
          </label>
          <a
            data-cy="doi-link"
            href="http://dx.doi.org/10.5067/GPMGV/LPVEX/MULTPLE/DATA202"
            rel="noopener noreferrer"
            style={
              Object {
                "color": "#aac9ff",
              }
            }
            target="_blank"
          >
            10.5067/GPMGV/LPVEX/MULTPLE/DATA202
          </a>
        </div>
        <div
          className="c6"
        >
          <div
            data-cy="data-product-campaigns"
          >
            <label
              data-cy="doi-campaign-label"
              style={
                Object {
                  "borderBottom": "1px solid #9e9e9e",
                  "color": "#9e9e9e",
                  "display": "block",
                }
              }
            >
              Campaigns
            </label>
            <a
              href="/campaign/ac9220f0-edde-47ee-ab0a-41e38221a104"
            >
              <small>
                Light Precipitation Evaluation Experiment
              </small>
            </a>
          </div>
          <div
            data-cy="data-product-platforms"
          >
            <label
              data-cy="doi-platform-label"
              style={
                Object {
                  "borderBottom": "1px solid #9e9e9e",
                  "color": "#9e9e9e",
                  "display": "block",
                }
              }
            >
              Platforms
            </label>
            <a
              href="/platform/704d3b91-7883-477c-be57-c50dd33f6165"
            >
              <small>
                University of Wyoming King Air
              </small>
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</section>
`;
